libs/features/seotech/seotech.js

```diff
@@ -21,14 +21,45 @@ export async function getVideoObject(url, seotechAPIUrl) {
   return body.videoObject;
 }
 
-export default async function appendVideoObjectScriptTag(url, { createTag, getConfig }) {
+export async function getStructuredData(url, sheetUrl, seotechAPIUrl) {
+  const apiUrl = new URL(seotechAPIUrl);
+  apiUrl.pathname = '/api/v1/web/seotech/getStructuredData';
+  apiUrl.searchParams.set('url', url);
+  if (sheetUrl) {
+    apiUrl.searchParams.set('sheetUrl', sheetUrl);
+  }
+  const resp = await fetch(apiUrl.href, { headers: { 'Content-Type': 'application/json' } });
+  const body = await resp?.json();
+  if (!resp.ok) {
+    throw new Error(`Failed to fetch structured data: ${body?.error}`);
+  }
+  return body.objects;
+}
+
+export async function appendScriptTag({ locationUrl, getMetadata, createTag, getConfig }) {
+  const windowUrl = new URL(locationUrl);
   const seotechAPIUrl = getConfig()?.env?.name === 'prod'
     ? SEOTECH_API_URL_PROD : SEOTECH_API_URL_STAGE;
-  try {
-    const obj = await getVideoObject(url, seotechAPIUrl);
+
+  const append = (obj) => {
     const script = createTag('script', { type: 'application/ld+json' }, JSON.stringify(obj));
     document.head.append(script);
-  } catch (e) {
-    logError(e.message);
+  };
+
+  const prs = [];
```

Comment:
prs?

Reply:
_Promises_, I guess

Reply:
Yes. Does this repo prefer I spell these  vars out?

Reply:
https://github.com/adobecom/milo/pull/1115/commits/40dc3fec84d0266e4294182fd506011b58e80fdb